Rakefile

```diff
@@ -73,7 +73,8 @@ GEM_INFO = {
     version_getter: ->() {
       require './lib/opentelemetry/adapters/concurrent-ruby/version.rb'
       OpenTelemetry::Adapters::ConcurrentRuby::VERSION
-    }
+    },
+    gem_dir: 'adapters/concurrent-ruby'
```

Comment:
It seems that the real issue is that there's a violation of naming conventions. The gem name `opentelemetry-adapters-concurrent_ruby` matches the namespace in the gem `OpenTelemetry::Adapters::ConcurrentRuby`, so it seems that's the intended name. However, the gem directory here doesn't match; the gem should be moved into `adapters/concurrent_ruby`. (Additionally, the require path _within_ the gem is also incorrect. It is `opentelemetry/adapters/concurrent-ruby` but should be `opentelemetry/adapters/concurrent_ruby`.)

Reply:
Thanks for the feedback on the best way to handle this @dazuma. I updated the folder layout to follow conventions.